activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -55,7 +55,13 @@ def self.initialize_i18n(app)
 
       reloader = ActiveSupport::FileUpdateChecker.new(I18n.load_path.dup){ I18n.reload! }
       app.reloaders << reloader
-      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated }
+      ActionDispatch::Reloader.to_prepare do
+        reloader.execute_if_updated
+        # TODO: remove the following line as soon as the return value of
+        # callbacks is ignored, that is, returning `false` does not
+        # display a deprecation warning or halts the callback chain.
+        true
```

Comment:
@rafaelfranca I also changed this as you suggested, so it has the TODO statement and simply returns true.